controllers/common/util.go

```diff
@@ -503,141 +503,35 @@ func GetApprovalModeinNs(r client.Reader, ns string) (approvalMode string, err e
 
 // GetCatalogSource gets CatalogSource will be used by operators
 func GetCatalogSource(packageName, ns string, r client.Reader) (CatalogSourceName, CatalogSourceNS string) {
-	pmList := &operatorsv1.PackageManifestList{}
-
-	if err := r.List(context.TODO(), pmList, &client.ListOptions{Namespace: ns}); err != nil {
+	
+	subList := &olmv1alpha1.SubscriptionList{}
+	if err := r.List(context.TODO(), subList, &client.ListOptions{Namespace: ns}); err != nil {
 		klog.Info(err)
 	}
 
-	var packageManifestList []operatorsv1.PackageManifest
-	for _, pm := range pmList.Items {
-		if pm.Status.PackageName != packageName {
+	var subscriptionList []olmv1alpha1.SubscriptionList
+	for _, sub := range subList.Items {
+		if sub.Spec.name != packageName {
 			continue
-		} else if pm.Status.CatalogSource == "redhat-operators" ||
-			pm.Status.CatalogSource == "redhat-marketplace" ||
-			pm.Status.CatalogSource == "community-operators" ||
-			pm.Status.CatalogSource == "certified-operators" {
+		} else if sub.Spec.name == "ibm-namespace-scope-operator" ||
+			sub.Spec.name == "ibm-odlm" {
 			continue
 		}
-		packageManifestList = append(packageManifestList, pm)
+		subscriptionList = append(subscriptionList, sub)
```

Comment:
Could we check `sub.Spec.name == packageName` and if it is yes, append it into `subscriptionList